runtime-light/stdlib/time/timelib-functions.h

```diff
@@ -4,14 +4,78 @@
 
 #pragma once
 
+#include <chrono>
+#include <cmath>
 #include <cstdint>
+#include <cstdlib>
+#include <expected>
+#include <format>
 #include <optional>
 #include <string_view>
 
 #include "kphp/timelib/timelib.h"
 
+#include "common/containers/final_action.h"
+#include "runtime-common/core/std/containers.h"
+#include "runtime-light/allocator/allocator.h"
+
 namespace kphp::timelib {
 
+constexpr inline std::array<std::string_view, 12> MON_FULL_NAMES = {"January", "February", "March",     "April",   "May",      "June",
+                                                                    "July",    "August",   "September", "October", "November", "December"};
+constexpr inline std::array<std::string_view, 12> MON_SHORT_NAMES = {"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
+constexpr inline std::array<std::string_view, 7> DAY_FULL_NAMES = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
+constexpr inline std::array<std::string_view, 7> DAY_SHORT_NAMES = {"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"};
+
+struct destructor {
+  void operator()(timelib_error_container* ec) const noexcept {
+    kphp::memory::libc_alloc_guard{}, timelib_error_container_dtor(ec);
+  }
+
+  void operator()(timelib_rel_time* rt) const noexcept {
+    kphp::memory::libc_alloc_guard{}, timelib_rel_time_dtor(rt);
+  }
+
+  void operator()(timelib_time* t) const noexcept {
+    kphp::memory::libc_alloc_guard{}, timelib_time_dtor(t);
+  }
+
+  void operator()(timelib_time_offset* to) const noexcept {
+    kphp::memory::libc_alloc_guard{}, timelib_time_offset_dtor(to);
+  }
+};
+
+using error_container_t = std::unique_ptr<timelib_error_container, destructor>;
+using rel_time_t = std::unique_ptr<timelib_rel_time, destructor>;
+using time_t = std::unique_ptr<timelib_time, destructor>;
+using time_offset_t = std::unique_ptr<timelib_time_offset, destructor>;
+
+time_offset_t construct_time_offset(timelib_time& t) noexcept;
+
+std::pair<time_t, error_container_t> parse_time(std::string_view time_sv) noexcept;
+std::pair<time_t, error_container_t> parse_time(std::string_view time_sv, const char* format) noexcept;
+std::expected<rel_time_t, std::format_string<const char*>> parse_interval(std::string_view format) noexcept;
+
+time_t add(timelib_time& t, timelib_rel_time& interval) noexcept;
+
+rel_time_t clone(timelib_rel_time& rt) noexcept;
```

Comment:
Isn't it better to just create a wrapper class? As far as I see, it might be worth it since you already have clone and destruct